### PR TITLE
Add fault injection unittest to increase coverage

### PR DESCRIPTION
**Pull request by brawner**
This redoes #48 and adds unit tests and fault injection macros to rcl_logging_spdlog. Coverage can get to 99% with this. I'll post CI tests soon.

### CODE REVIEW

**Comment by brawner** [comment]
Builds and tests of the following packages `rcutils`, `rcl_logging_spdlog`, `rosidl_typesupport_c`, `rosidl_typesupport_cpp`, `rcl`, `rcl_action`, `rcl_lifecycle`, `rosidl_runtime_c`, `rmw_dds_common`, `rmw`

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11916/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6961/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9709/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/11850/)

**Comment by brawner** [comment]
Testing this PR branch against current ROS 2 branches.

Testing `--packages-select rcl_logging_spdlog`
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12054/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7077/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9827/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/11987/)

File: rcl_logging_spdlog/test/test_logging_interface.cpp [hunk 1]
```diff
@@ -162,3 +163,16 @@ TEST_F(LoggingTest, full_cycle)
     expected_log.str(),
     actual_log.str()) << "Unexpected log contents in " << log_file_path;
 }
+
+TEST_F(LoggingTest, init_fini_maybe_fail_test)
+{
+  RCUTILS_FAULT_INJECTION_TEST(
+  {
+    if (RCL_LOGGING_RET_OK == rcl_logging_external_initialize(nullptr, allocator)) {
+      EXPECT_EQ(RCL_LOGGING_RET_OK, rcl_logging_external_shutdown());
```
Comment (hidmic) [inline rcl_logging_spdlog/test/test_logging_interface.cpp:172]: @brawner what if `rcl_logging_external_shutdown()` fails because of fault injection?

Comment (brawner) [reply rcl_logging_spdlog/test/test_logging_interface.cpp:172]: `rcl_logging_external_shutdown()` is currently very simple and doesn't call any external functions. The `EXPECT_EQ` will help notify people this test needs to be fixed should that function be modified.

https://github.com/ros2/rcl_logging/blob/master/rcl_logging_spdlog/src/rcl_logging_spdlog.cpp#L150-L154

File: rcl_logging_spdlog/test/test_logging_interface.cpp [hunk 2]
```diff
@@ -17,6 +17,7 @@
 #include <rcutils/allocator.h>
 #include <rcutils/env.h>
 #include <rcutils/error_handling.h>
+#include <rcutils/testing/fault_injection.h>
 #include <rcutils/logging.h>
```
[the remark this answers is not in the document]
Comment (brawner) [reply rcl_logging_spdlog/test/test_logging_interface.cpp]: Fixed.

**Comment by brawner** [comment]
Once more unto the breach dear friends, once more

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12074/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7097/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9847/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12003/)

**Comment by brawner** [comment]
rpr build fails because rcutils needs a new release in order for the fault injection macros to be included.

**Comment by brawner** [comment]
CI builds all pass though, so this is good

**Pull request merged**